src/arrayMethodJoin.js

```diff
@@ -5,7 +5,33 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    const modifiedThis = [...this];
+    const newArray = [];
+    let result = '';
+
+    for (let i = 0; i < modifiedThis.length; i++) {
+      if (modifiedThis[i] === null || modifiedThis[i] === undefined) {
+        modifiedThis[i] = '';
+      }
+    }
+
+    for (let i = 0; i < modifiedThis.length; i++) {
+      newArray[i * 2] = modifiedThis[i];
+    }
+
+    for (let i = 1; i < newArray.length; i = i + 2) {
+      newArray[i] = separator;
+
+      if (separator === undefined) {
+        newArray[i] = ',';
+      }
+    }
+
+    for (const element of newArray) {
+      result += element;
+    }
+
+    return result;
```

Comment:
do you really need 4 loops to solve the task? actually you can do it using just 1 loop